### ISSUE TITLE
Resolve Failing Topgun Tests on PKS

### ISSUE THREAD

**Issue by scottietremendous** [concourse/concourse#4098]
[@cirocosta here]:

**Summary**

In our attempts to run Concourse with PKS, we've encountered a number of failing
[topgun tests][topgun]. 

At the moment, there are two classes of tests that fail:

- [baggageclaim drivers](https://github.com/concourse/concourse/blob/2c3e9dd527a332bd5816228ff67596da5bcf791f/topgun/k8s/baggageclaim_drivers_test.go)
- [container limits](https://github.com/concourse/concourse/blob/2c3e9dd527a332bd5816228ff67596da5bcf791f/topgun/k8s/container_limits_test.go)

In order the have [PKS tests in the main pipeline](https://github.com/concourse/ci/issues/40) successfully running, these
need to be addressed first.

**Acceptance Criteria**

- [ ] All failing tests passing.

**Extra context**

Those two classes of test have something in common - they require knowledge of
what is the underlying infrastructure that they run on.

E.g., to ensure that a particular set of baggageclaim drivers work across a
given set of node-level base images, we intentionally target a set of node pools
that contain a specific tag.

The number of pools to target then varies greatly:

- GKE allows us to have 3 different types of base images:
  - container optimized
  - container optimized (COS) with ContainerD
  - ubuntu
- PKS allows us to use only one - the Ubuntu-based stemcells.

[topgun]: https://github.com/concourse/concourse/tree/master/topgun/k8s

Thank you!

**Comment by cirocosta** [comment]
List of failing tests:

```
Summarizing 8 Failures:

[Fail] Container Limits [JustBeforeEach] using cos as NodeImage returns the configure default container limit
/Users/pivotal/workspace/concourse/topgun/k8s/k8s_suite_test.go:172

[Fail] Container Limits [JustBeforeEach] using Ubuntu as NodeImage fails to set the memory limit
/Users/pivotal/workspace/concourse/topgun/k8s/k8s_suite_test.go:172

[Fail] Baggageclaim Drivers across different node images [It] with btrfs on cos
/Users/pivotal/workspace/concourse/topgun/k8s/k8s_suite_test.go:172

[Fail] Baggageclaim Drivers across different node images [It] with btrfs on ubuntu
/Users/pivotal/workspace/concourse/topgun/k8s/k8s_suite_test.go:172

[Fail] Baggageclaim Drivers across different node images [It] with overlay on cos
/Users/pivotal/workspace/concourse/topgun/k8s/k8s_suite_test.go:172

[Fail] Baggageclaim Drivers across different node images [It] with overlay on ubuntu
/Users/pivotal/workspace/concourse/topgun/k8s/k8s_suite_test.go:172

[Fail] Baggageclaim Drivers across different node images [It] with naive on cos
/Users/pivotal/workspace/concourse/topgun/k8s/k8s_suite_test.go:172

[Fail] Baggageclaim Drivers across different node images [It] with naive on ubuntu
/Users/pivotal/workspace/concourse/topgun/k8s/k8s_suite_test.go:172

Ran 40 of 40 Specs in 1248.760 seconds
FAIL! -- 32 Passed | 8 Failed | 0 Pending | 0 Skipped
```

**Comment by cirocosta** [comment]
Hey, @taylorsilva and I started working on a PR to address those 8 tests that currently fail - see https://github.com/concourse/concourse/pull/4102.

**Comment by taylorsilva** [comment]
We got all failing tests passing. When we tried running the entire test suite we started getting random failures from other tests not part of baggageclaim and containerLimits. We decided to scale up vertically the size of the worker nodes in the cluster to see if that stopped these flakes from happening.

Current worker nodes size: 2 CPU, 4GB RAM

on GKE, worker nodes size: 2 CPU, 8GB RAM

The new PKS cluster should have 2 CPU and 8GB RAM now.
